incubator/hnc/internal/validators/object.go

```diff
@@ -106,8 +107,13 @@ func (o *Object) handle(ctx context.Context, log logr.Logger, op admissionv1beta
 	oldSource, oldInherited := metadata.GetLabel(oldInst, api.LabelInheritedFrom)
 	newSource, newInherited := metadata.GetLabel(inst, api.LabelInheritedFrom)
 
-	// If the object wasn't and isn't inherited, it's none of our business.
+	// If the object wasn't and isn't inherited, we will check to see if the
+	// source can be created without causing any conflict.
 	if !oldInherited && !newInherited {
+		if yes, cnses := o.hasConflict(inst); yes {
+			msg := fmt.Sprintf("Cannot create %s '%s' in namespace '%s' because it would overwrite objects in the following descendant namespace(s): %v. Choose a different name for the object, or remove the conflicting objects from these namespaces.", inst.GroupVersionKind(), inst.GetName(), inst.GetNamespace(), cnses)
```

Comment:
Nit: use `%q` instead of `'%s'`

Reply:
Also, it might be nice to introduce newlines, e.g.:

```
Cannot create %s '%s' in namespace '%s' because it would overwrite objects in the following descendant namespace(s):
  * NS 1
  * NS 2
Choose a different name for the object, or remove the conflicting objects from these namespaces.
```

This will make the message _much_ easier to read, IMO, especially the last line.

Reply:
sg, will update the fmt in #1092, sgty?

Reply:
sgtm